### PR TITLE
Integrate Sequence Parallelism branches

### PR DESCRIPTION
**Pull request by l-yohai**
## Title

- Sequence parallelism (feat. @reniew, @ohwi, @l-yohai)

## Description

- This PR is Integration of SP current version. But there is something wrong.
- We will fix the bugs for the coming week and write test modules according to the SP design.

* It did not include the contents of the branch that worked for the test.

### CODE REVIEW

**Comment by hyunwoongko** [comment]
@l-yohai when you are ready. please notice me :)

**Comment by l-yohai** [comment]
@hyunwoongko I'm ready. But pre-commit of other files except for sp fails again and again... 😂

**Comment by hyunwoongko** [comment]
@l-yohai pre-commit is ok lol
nervermind.

File: oslo/torch/nn/parallel/distributed/data_parallel/_sequence_parallel.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import torch
```
Comment (hyunwoongko) [inline oslo/torch/nn/parallel/distributed/data_parallel/_sequence_parallel.py:1]: @l-yohai how about changing this file to `sequence_parallel.py`?
I mean removing `_` prefix.

File: tests/test_sp_multiheadattention.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import torch
```
Comment (hyunwoongko) [inline tests/test_sp_multiheadattention.py:1]: let's move this `tests/nn/data_parallel/test_sp_m...py`.

File: oslo/torch/nn/parallel/distributed/data_parallel/_sequence_parallel.py (hunk 1 above)
Comment (l-yohai) [reply oslo/torch/nn/parallel/distributed/data_parallel/_sequence_parallel.py]: yes, I removed `_` prefix.

File: tests/test_sp_multiheadattention.py (hunk 2 above)
Comment (l-yohai) [reply tests/test_sp_multiheadattention.py]: I moved this file in `tests/nn/data_parallel`

**Pull request merged**